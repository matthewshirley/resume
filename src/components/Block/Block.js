import React from 'react';
import PropTypes from 'prop-types';

import Box from '../common/Box';
import Text from '../common/Text';
import Grid from '../common/Grid';

const Block = ({ title, children, borderBottom }) => (
  <Grid
    marginBottom={4}
    gridArea="content"
    gridTemplateColumns={{
      sm: '1fr',
      lg: '1fr 1fr 1fr',
      print: '1fr 1fr 1fr',
    }}
    gridTemplateAreas={{
      sm: '"RowTitle" "RowContent"',
      lg: '"RowTitle RowContent RowContent"',
      print: '"RowTitle RowContent RowContent"',
    }}
    borderBottom={borderBottom}
  >
    <Box gridArea="RowTitle">
      <Text as="h2" color="primary" fontSize={2} m={0} mb={{ sm: 2, lg: 0, print: 0 }}>{title}</Text>
    </Box>
    <Box gridArea="RowContent">
      {children}
    </Box>
  </Grid>
);

Block.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.element.isRequired,
  borderBottom: PropTypes.bool,
};

Block.defaultProps = {
  borderBottom: false,
};

export default Block;
