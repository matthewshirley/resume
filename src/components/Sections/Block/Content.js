import React from 'react';
import PropTypes from 'prop-types';

import { Text } from '@mattshirley/design/src/components/Text';
import { Box } from '@mattshirley/design/src/components/Layout/Box';

const Content = ({ summary }) => (
  <Box>
    <Text fontSize={1}>{summary}</Text>
  </Box>
);

Content.propTypes = {
  summary: PropTypes.string.isRequired,
};

export default Content;
