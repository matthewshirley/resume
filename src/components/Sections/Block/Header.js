import React from 'react';
import PropTypes from 'prop-types';

import { Text } from '@mattshirley/design/src/components/Text';
import { Box } from '@mattshirley/design/src/components/Layout/Box';
import { Flex } from '@mattshirley/design/src/components/Layout/Flex';

const Header = ({
  company, position, website, location, startDate, endDate,
}) => {
  const formattedStartDate = startDate;
  const formattedEndDate = endDate || 'Present';

  return (
    <Flex justifyContent="space-between" mb={2}>
      <Box>
        <Text fontSize={1} bold>
          {position}
        </Text>
        <Text as="small" fontSize={0}>
          <a href={website} target="_blank" rel="noopener noreferrer">
            {company}
          </a>
          {location && ` - ${location}`}
        </Text>
      </Box>
      <Text fontSize={0}>
        {formattedStartDate}
        {' '}
        -
        {' '}
        {formattedEndDate}
      </Text>
    </Flex>
  );
};

Header.propTypes = {
  company: PropTypes.string.isRequired,
  position: PropTypes.string.isRequired,
  website: PropTypes.string.isRequired,
  startDate: PropTypes.string.isRequired,
  endDate: PropTypes.oneOf([
    PropTypes.bool, PropTypes.string]).isRequired,
  location: PropTypes.string,
};

Header.defaultProps = {
  location: '',
};

export default Header;
