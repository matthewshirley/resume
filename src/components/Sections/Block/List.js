import React from 'react';
import PropTypes from 'prop-types';

import { Text } from '@mattshirley/design/src/components/Text';
import { Box } from '@mattshirley/design/src/components/Layout/Box';

const List = ({
  highlights,
}) => {
  const listOfHighlights = highlights.map((highlight, index) => (
    <Text key={index} as="li" fontSize={1} my={2}>{highlight}</Text>
  ));

  return (
    <Box as="ul" mt={2}>
      {listOfHighlights}
    </Box>
  );
};

List.propTypes = {
  highlights: [
    PropTypes.string.isRequired,
  ].isRequired,
};

export default List;
