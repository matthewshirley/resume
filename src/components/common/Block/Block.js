import React from 'react';
import PropTypes from 'prop-types';

import Box from '../Box';
import Text from '../Text';
import Grid from '../Grid';

const Block = ({ title, children, borderBottom }) => (
  <Grid
    marginBottom={4}
    gridArea="content"
    gridTemplateColumns={{
      sm: '1fr',
      lg: '1fr 1fr 1fr',
      print: '1fr 1fr 1fr',
    }}
    gridTemplateAreas={{
      sm: '"RowTitle" "RowContent"',
      lg: '"RowTitle RowContent RowContent"',
      print: '"RowTitle RowContent RowContent"',
    }}
    borderBottom={borderBottom}
  >
    <Box gridArea="RowTitle">
      <Text color="primary" fontSize={3} m={0} mb={{ sm: 2, lg: 0, print: 0 }}>{title}</Text>
    </Box>
    <Box gridArea="RowContent">
      {children}
    </Box>
  </Grid>
);

Block.propTypes = {
  title: PropTypes.string,
  borderBottom: PropTypes.bool,
  children: PropTypes.element.isRequired,
};

Block.defaultProps = {
  title: '',
  borderBottom: false,
};

export default Block;
