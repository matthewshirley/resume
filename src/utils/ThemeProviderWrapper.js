import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider } from 'styled-components';

import theme from '../theme';

/**
 * ThemeProvider for application and test specs
 *
 * @param {object} props
 * @param {element} props.children
 */
const ThemeProviderWrapper = ({ children }) => (
  <ThemeProvider theme={theme}>
    { children }
  </ThemeProvider>
);

ThemeProviderWrapper.propTypes = {
  children: PropTypes.element,
};

ThemeProviderWrapper.defaultProps = {
  children: null,
};

export default ThemeProviderWrapper;
