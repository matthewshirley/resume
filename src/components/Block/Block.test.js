import React from 'react';
import renderer from 'react-test-renderer';

import Block from './Block';
import ThemeProviderWrapper from '../../utils/ThemeProviderWrapper';

describe('Block', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Block title="Test Block" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with children', () => {
    const tree = renderer
      .create(<Block title="Test Block"><p>Test Children</p></Block>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with a border', () => {
    const tree = renderer
      .create(
        <ThemeProviderWrapper>
          <Block title="Test Block" borderBottom />
        </ThemeProviderWrapper>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
