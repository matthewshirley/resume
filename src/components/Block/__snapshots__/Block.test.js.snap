// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Block renders correctly 1`] = `
<div
  className="sc-AxjAm sc-AxhCb bNLhIJ"
>
  <div
    className="sc-AxjAm fIbOII"
  >
    <h2
      className="sc-AxirZ jaRDmo"
      color="primary"
      fontSize={2}
    >
      Test Block
    </h2>
  </div>
  <div
    className="sc-AxjAm jBohcl"
  />
</div>
`;

exports[`Block renders correctly with children 1`] = `
<div
  className="sc-AxjAm sc-AxhCb bNLhIJ"
>
  <div
    className="sc-AxjAm fIbOII"
  >
    <h2
      className="sc-AxirZ jaRDmo"
      color="primary"
      fontSize={2}
    >
      Test Block
    </h2>
  </div>
  <div
    className="sc-AxjAm jBohcl"
  >
    <p>
      Test Children
    </p>
  </div>
</div>
`;

exports[`Block renders with a border 1`] = `
<div
  className="sc-AxjAm sc-AxhCb eOIZoi"
>
  <div
    className="sc-AxjAm fIbOII"
  >
    <h2
      className="sc-AxirZ cXGILF"
      color="primary"
      fontSize={2}
    >
      Test Block
    </h2>
  </div>
  <div
    className="sc-AxjAm jBohcl"
  />
</div>
`;
